src/jit/gentree.cpp

```diff
@@ -14397,7 +14397,7 @@ bool Compiler::gtTreeHasSideEffects(GenTreePtr tree, unsigned flags /* = GTF_SID
     return true;
 }
 
-GenTreePtr Compiler::gtBuildCommaList(GenTreePtr list, GenTreePtr expr)
+GenTreePtr Compiler::gtBuildCommaList(GenTreePtr list, GenTreePtr expr DEBUGARG(bool remorphing))
```

Comment:
It is probably a matter of taste, but I think that we are better off with a purer invariant (e.g. that the VNs are either guaranteed to be preserved of not) and in any case, I don't think the value of the assertion justifies the additional clutter to the code. But again it is just an opinion.